### ISSUE TITLE
Firefox hashes mismatch sometimes

### ISSUE THREAD

**Issue by grahamc** [mozilla/nixpkgs-mozilla#65]
I get:

output path ‘/nix/store/hc46q7rs43aphmdcf6gyvnwj4msmgldd-firefox-57.0a1.en-US.linux-x86_64.tar.bz2’ has sha512 hash ‘2r8m2k0aw844xd125yqs83qblh645sk0zq9rvmzvv8kjrzkivshzpfv1xg2rlijar819l9k1ics5ifr4acj4f9103dvfpmhqk1fii2f’ when ‘1wrzqadr4k77gpl78v99ijzg5p9j6fsndsr42kxw84dgl22yanyz1md7p3vsf8m2fq93nfs8zksnnm6jj66r92b9qchjz09xfnnaqd3’ was expected

and opened https://bugzilla.mozilla.org/show_bug.cgi?id=1396337 to track it on releng side of things.

In short, trying several hours later fixes it.

**Comment by rail** [comment]
Yeah, I get that too from time to time... The reason is that we fetch the tarball from a stable URL, while it can be overwritten multiple times a day (usually twice).

Maybe we should use https://download.cdn.mozilla.net/pub/firefox/nightly/latest-mozilla-central/firefox-65.0a1.en-US.linux-x86_64.buildhub.json to determine the URL of the tarball pointing to a dated directory.
So instead of using the [latest](https://github.com/mozilla/nixpkgs-mozilla/blob/0d64cf67dfac2ec74b2951a4ba0141bc3e5513e8/firefox-overlay.nix#L60) directory we could calculate something similar to [this](https://github.com/mozilla/nixpkgs-mozilla/blob/0d64cf67dfac2ec74b2951a4ba0141bc3e5513e8/firefox-overlay.nix#L61).

**Comment by rail** [comment]
I found an [issue](https://github.com/mozilla-releng/beetmoverscript/issues/195) with the file I mentioned above. Until it's fixed [this hack](https://gist.github.com/ff209cd6381f73b038775b94f8aa56eb) should work. I can update the patch after the bug is fixed.

**Comment by rail** [comment]
the hack worked fine for me today:
```
building '/nix/store/3c8gqikqc08928gywkwd901s9by2762x-firefox-66.0a1.en-US.linux-x86_64.tar.bz2.drv'...

trying https://archive.mozilla.org/pub/firefox/nightly/2018/12/2018-12-13-09-42-43-mozilla-central/firefox-66.0a1.en-US.linux-x86_64.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 66.3M  100 66.3M    0     0  14.6M      0  0:00:04  0:00:04 --:--:-- 14.6M
```

So just waiting for the blocking issue to be fixed, then I'll just adjust the patch.
